WMF Framework/WMFExploreFeedContentController.m

```diff
@@ -534,29 +534,19 @@ - (void)toggleGlobalContentGroupKinds:(BOOL)on {
         }
         [newPreferences setObject:newGlobalCardPreferences forKey:WMFExploreFeedPreferencesGlobalCardsKey];
         return newPreferences;
-    } willTurnOnContentGroupOrLanguage:on waitForCallbackFromCoordinator:YES apply:YES updateFeed:YES completion:nil];
+    } willTurnOnContentGroupOrLanguage:on waitForCallbackFromCoordinator:YES apply:YES updateFeed:updateFeed completion:nil];
 }
 
 - (void)saveNewExploreFeedPreferences:(NSDictionary *)newExploreFeedPreferences apply:(BOOL)apply updateFeed:(BOOL)updateFeed {
-    WMFAsyncBlockOperation *op = [[WMFAsyncBlockOperation alloc] initWithAsyncBlock:^(WMFAsyncBlockOperation *_Nonnull op) {
-        dispatch_async(dispatch_get_main_queue(), ^{
-            NSManagedObjectContext *moc = self.dataStore.feedImportContext;
-            [moc performBlock:^{
-                [moc wmf_setValue:newExploreFeedPreferences forKey:WMFExploreFeedPreferencesKey];
-                if (apply) {
-                    [self applyExploreFeedPreferencesToAllObjectsInManagedObjectContext:moc];
-                }
-                [self save:moc];
-                dispatch_async(dispatch_get_main_queue(), ^{
-                    if (updateFeed) {
-                        [self updateFeedSourcesUserInitiated:NO completion:nil];
-                    }
-                    [op finish];
-                });
-            }];
-        });
-    }];
-    [self.operationQueue addOperation:op];
+    WMFAssertMainThread(@"Saving explore feed preferences should be performed on the main thread");
```

Comment:
Makes sense, I would prefer to keep it on the `feedImportContext` but I was worried that there could be a scenario in which the user applies the preferences in settings -> update preferences operation is added to the operation queue -> user leaves settings -> user remembers that they wanted to apply another preference -> user goes back to settings -> settings are off because the operation wasn't executed yet.

I think the completion in `updateExploreFeedPreferences` was meant to be called there and not go all the way to `saveNewExploreFeedPreferences` - what bug was it causing?

Reply:
What if we saved the preferences on the main thread but applied them on the feedImportContext?

Reply:
> I think the completion in updateExploreFeedPreferences was meant to be called there and not go all the way to saveNewExploreFeedPreferences - what bug was it causing?

There wasn't a bug, so if that wasn't the intent, that's fine. It might be worth renaming the completion to clarify that it doesn't indicate that the preferences have been fully applied

> What if we saved the preferences on the main thread but applied them on the feedImportContext?

That's a good idea